doc/src/modules/combinatorics/fp_groups.rst

```diff
@@ -0,0 +1,185 @@
+
+
+Construction of words
+---------------------
+
+Can be called with different public versions of ``FreeGroup`` i.e
+``free_group``, ``vfree_group`` and ``xfree_group``.
+
+Methods for manipulating Words
+------------------------------
+
+This section describes some basic access functions for words. These operations apply
+to both, free groups and arbitrary fp-groups.
+
+copy
+return a
+
+is_identity
+
+array_form
+
+letter_form
+
+index
+
+ext_rep
+
+contains
+
+eliminate_word
+
+len
+
+``in``
+
+``len(w)``
+The lenth of word ``w``.
+
+
+Comparison of words
+-------------------
+
+Coset Enumeration: The Todd-Coxeter Algorithm
+=============================================
+
+This section describes the use of coset enumeration techniques in SymPy. The
+algorithm used for coset enumeration procedure is Todd-Coxeter algorithm and
+is developed in Sympy using [Ho05] and [CDHW73]. The reader should consult
+[CDHW73] and [Hav91] for a general description of the algorithm.
+
+CosetTable
+----------
+
+Class used to manipulate the information regarding the coset enumeration of
+the finitely presented group ``G`` on the cosets of the subgroup ``H``.
+
+Definition
+----------
+
+Basically a coset table is the permutation representation of the finitely
+presented group on the cosets of a subgroup. Most of the set theoretic and
+group functions use the regular representation of ``G``, i.e., the coset table
+of ``G`` over the trivial subgroup.
+
+The actual mathematical coset table is obtained using ``.table`` and is a list
+of lists. For each generator of ``G`` and its inverse the table contains a
+coloumn. Each coloumn is simply a list of integers. If ``l`` is the generator
+list for the generator `g` and if ``l[i] = j`` then generator `g` takes the
+coset `i` to the coset `j` by multiplication from the right.
+
+For finitely presented groups, a coset table is computed by a Todd-Coxeter
+coset enumeration. Note that you may influence the performance of that
+enumeration by changing the values of the variable
+``CosetTable.coset_table_max_limit``.
+
+For experienced users we have a number of parameters that can be used to
+manipulate the algorithm, like
+
+* ``coset_table_max_limit``: manipulate the maximum number of cosets allowed
+  in coset enumeration. A coset enumeration will not finish if the subgroup
+  does not have finite index, and even if it has it may take many more
+  intermediate cosets than the actual index of the subgroup is. To avoid a
+  coset enumeration "running away" therefore SymPy has a "safety stop"
+  built-in. This is controlled by this variable.
+
+* ``max_stack_size``: manipulate the maximum size of ``deduction_stack`` above
+  to equal to which the stack is emptied.
+
+Compression and Standardization
+-------------------------------
+
+
+Subgroups of Finite Index: Low Index Subgroups algorithm
+========================================================
+
+Bibliography
```

Comment:
I think it could be left here for now. There is currently no `combinatorics/literature.rst`.